controllers/bpfman-agent/common.go

```diff
@@ -240,15 +243,56 @@ func interfaceInExcludeList(interfaceSelector *bpfmaniov1alpha1.InterfaceSelecto
 	return false
 }
 
+func setupAllowedInterfacesLists(interfaceSelector *bpfmaniov1alpha1.InterfaceSelector) ([]*regexp.Regexp, []string) {
+	var isRegexp = regexp.MustCompile("^/(.*)/$")
+	var allowedRegexpes []*regexp.Regexp
+	var allowedMatches []string
+
+	for _, definition := range interfaceSelector.InterfacesDiscoveryConfig.AllowedInterfaces {
+		definition = strings.Trim(definition, " ")
+		// the user defined a /regexp/ between slashes: compile and store it as regular expression
+		if sm := isRegexp.FindStringSubmatch(definition); len(sm) > 1 {
+			re, err := regexp.Compile(sm[1])
+			if err != nil {
+				return allowedRegexpes, allowedMatches
+			}
+			allowedRegexpes = append(allowedRegexpes, re)
+		} else {
+			// otherwise, store it as exact match definition
+			allowedMatches = append(allowedMatches, definition)
+		}
+	}
+	return allowedRegexpes, allowedMatches
+}
+
+func interfaceInAllowedList(intf string, allowedRegexpes []*regexp.Regexp, allowedMatches []string) bool {
+	if len(allowedRegexpes) == 0 && len(allowedMatches) == 0 {
+		return true
+	}
+
+	for _, re := range allowedRegexpes {
+		if re.MatchString(intf) {
+			return true
+		}
+	}
+
+	for _, n := range allowedMatches {
+		if n == intf {
+			return true
+		}
+	}
+	return false
+}
+
 func getInterfaces(interfaceSelector *bpfmaniov1alpha1.InterfaceSelector, ourNode *v1.Node, discoveredInterfaces *sync.Map) ([]string, error) {
```

Comment:
I was thinking we could define something like:
```
type netnsInterface struct {
	interfaceName string
	netnsPath     string
}
```
And then have `getInterfaces()` return `[]netnsInterface`.  The code that handles interface discovery could populate netnsPath as appropriate as it goes through the loop.  If an interface isn't in a network namespace, then we could set `netnsPath` to `""`.  This would be the case for all of the interfaces returned if interface discovery isn't enabled.  What do you think?

Reply:
Just thinking some more, we might want to have a different function for this calle `getDiscoveredInterfaces()` for this, and have `getInterfaces()` still return `[]string`.